targets/esp8266/jswrap_esp8266.c

```diff
@@ -87,13 +82,8 @@ __attribute__((section(".force.text"))) void jswrap_ESP8266_neopixelWrite(JsVar
     jsExceptionHere(JSET_ERROR, "Data length must multiples of RGB bytes (3).");
     return;
   }
-  uint8_t *pixels = (uint8 *)os_malloc(dataLength);
-  if (pixels == NULL) {
-    jsExceptionHere(JSET_ERROR, "Out of memory.");
-    return;
-  }
-  int i;
-  for (int i=0; i<dataLength; i++) {
+
+  for (size_t i=0; i<dataLength; i++) {
     JsVar *jsItem = jsvGetArrayItem(jsArrayOfData, i);
```

Comment:
You don't need any of this now... `JSV_GET_AS_CHAR_ARRAY` gets the char array but *also makes sure it's filled with the correct data*